dvc/fs/base.py

```diff
@@ -179,10 +179,11 @@ def open(
         self,
         path: AnyFSPath,
         mode: str = "r",
-        encoding: Optional[str] = None,
         **kwargs,
     ) -> "IO":  # pylint: disable=arguments-differ
-        return self.fs.open(path, mode=mode, encoding=encoding, **kwargs)
+        if "b" in mode:
+            kwargs.pop("encoding", None)
```

Comment:
Hm, why do we even need to pop it? In the #7691 we are not even getting the encoding here.

Reply:
`RepoFS` and `DvcFS` always sends `encoding=None` to the filesystems even in binary mode. `open()` is not fsspec compliant and can be fixed at that time in other filesystems. For now, I think this is the simplest approach.

Reply:
Let's fix repofs/dvcfs then if that is not too much work. Seems like it is a simple arg adjustment. Unless I'm missing something.

Reply:
I'd still try to avoid this bug here even if we fixed on repofs/dvcfs, until we fix this on `fsspec`'s side.